### PR TITLE
gh-102011: use sys.exception() instead of sys.exc_info() in docs where possible

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-102011

### CODE REVIEW

File: Doc/library/traceback.rst [hunk 1]
```diff
@@ -17,8 +17,7 @@ interpreter.
 .. index:: object: traceback
 
 The module uses traceback objects --- this is the object type that is stored in
-the :data:`sys.last_traceback` variable and returned as the third item from
-:func:`sys.exc_info`.
+the :data:`sys.last_traceback` variable and returned :func:`sys.exception`.
```
Comment (erlend-aasland) [inline Doc/library/traceback.rst]: Returned by?

```suggestion
the :data:`sys.last_traceback` variable and returned by :func:`sys.exception`.
```

File: Doc/library/traceback.rst [hunk 2]
```diff
@@ -17,8 +17,7 @@ interpreter.
 .. index:: object: traceback
 
 The module uses traceback objects --- this is the object type that is stored in
-the :data:`sys.last_traceback` variable and returned as the third item from
-:func:`sys.exc_info`.
+the :data:`sys.last_traceback` variable and returned by :func:`sys.exception`.
```
Comment (TeamSpen210) [inline Doc/library/traceback.rst]: Looks like this is missing `.__traceback__`?

**Review by erlend-aasland — APPROVED** [review]

**Comment by CAM-Gerlach** [comment]
Should we backport this to 3.11, since `sys.exception` was introduced in that version? Or is there a reason this needs to stay 3.12 only?

**Comment by erlend-aasland** [comment]
SGTM, @CAM-Gerlach

**Pull request merged**